### PR TITLE
24 3.4 notes on applications additional tests

### PR DESCRIPTION
**Pull request by aloeser**
Added tests for routing and abilities.

### CODE REVIEW

File: spec/models/ability_spec.rb [hunk 1]
```diff
@@ -71,6 +71,22 @@
     end
   end
 
+  %i[tutor organizer].each do |role|
+    it "can view and add notes to application letters" do
```
Comment (frederike-ramin) [inline spec/models/ability_spec.rb]: This doesn't seem to be recommended as it was not used in the other tests above. (also "{role}" is not that meaningful, is it?)

Comment (frederike-ramin) [inline spec/models/ability_spec.rb]: @querenker Would you approve that in that case?

Comment (querenker) [inline spec/models/ability_spec.rb]: @frederike-ramin Sorry for the late response (and the missing hash symbol). I meant
`it "can view and add notes to application letters as #{role}" do`. If one out of the two checked roles are failing the test, you will get this error message:

>  1) User can view and add notes to application letters
>       Failure/Error: expect(ability).to be_able_to(:view_and_add_notes, ApplicationLetter)
>         expected to be able to view_and_add_notes ApplicationLetter
>         ./spec/models/ability_spec.rb:79:in `block (3 levels) in <top (required)>'

Now I don't know which of the two roles is responsible for this error. If both role abilities are wrong you will get two times exactly the same error message (which is also weird for an other developer). With the change the error message would include the role which is responsible for the error.

>  1) User can view and add notes to application letters as tutor
>       Failure/Error: expect(ability).to be_able_to(:view_and_add_notes, ApplicationLetter)
>         expected to be able to view_and_add_notes ApplicationLetter
>         ./spec/models/ability_spec.rb:79:in `block (3 levels) in <top (required)>'

It is not a major issue and it concerns other (older) tests written like that, but I think it is nice and can maybe help other developers one day? :)

Comment (frederike-ramin) [inline spec/models/ability_spec.rb]: I see what you mean. I didn't expect the {role} to evaluate to the actual role, so I didn't get your point. I will change that.

**Review by querenker — APPROVED** [review]

**Comment by ekrebs5** [comment]
I also approve this :)

**Review by ekrebs5 — APPROVED** [review]

**Pull request merged**